xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-ui/src/main/resources/Main/SolrSearchConfig.xml

```diff
@@ -51,6 +51,13 @@
     'OBJECT': 'objcontent',
     'OBJECT_PROPERTY': 'propertyvalue'
   },
+  'phraseFields': 'title_en^200.0 title__^200.0
+                   filename^45.0 
+                   doccontent_en^40.0 doccontent__^40.0
+                   attcontent_en^35.0 attcontent__^35.0 
+                   comment^25.0
+                   author_display^20.0 creator_display^20.0  attauthor_display^20.0'
+  ,
```

Comment:
I prefer to have the phrase fields set by result type, as we do with the query fields.

Reply:
Can't we use the value of the query fields to set the phrase fields? i.e. have by default query fields = phrase fields. This means we don't specify phrase fields in the default configuration but we check for it below so that users can have different phrase fields than query fields.

Reply:
Marius, what do you mean by result type exactly? I tried to group as you did.

I did try query fields = phrase fields initially, but by gradual experimentation found that some fields were skewing my results the wrong way. For example *_raw would boost documents with a large number of links that weren't as relevant as the docs they pointed too. Maybe we could set the weighting to be similar, I wasn't sure if the qf score was summed with the pf score hence why I went for large numbers so that pf would always trump qf. Furthermore some of the shorter fields like author didn't really make sense to me as places to do phrase matching.

Reply:
OK, so indeed it makes more sense to have separate values (fields and weights) between ``qf`` and ``pf``. By "result type" I meant "document", "attachment", "object", "object property". So I meant having separate ``pf`` value when searching for documents than when searching for attachments. For instance attachment rows in the index don't have the ``doccontent`` field so it doesn't make sense to list it in the ``pf`` value when you look for attachments.

Reply:
> I meant "document", "attachment", "object", "object property"
Ah! I see... I did try this but had to merge into a single string for some reason. I'll try and figure it out and post back. Believe it had something to do with the fact that when searching across all result types the pf type parameters weren't being concatenated. 

On the subject of result types. I couldn't select combinations of result types from the facet menu, e.g. document and attachment. Is this behaviour by design?

Reply:
> Is this behaviour by design?

Yes, see this [comment](https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-ui/src/main/resources/Main/SolrSearchMacros.xml#L657). In order to have multiple selection we would need to use the same query fields for all result types. We don't do this because some fields are shared between results types and their importance (weight) depends on the result type. For instance the ``filename`` is very important when searching for attachments and less important when searching for documents. But if you search for both attachments and documents then the ``filename`` should have the weight close to the weight of the document name.

Reply:
> By "result type" I meant "document", "attachment", "object", "object property". So I meant having separate pf value when searching for documents than when searching for attachments

@mflorea, if I convert this to be by result type, how do I get it to also work when searching across all entities. As mentioned previously,

> when searching across all result types the pf type parameters weren't being concatenated.

If I implement as suggested we would lose all `pf` bosting when unselecting the facets. I think there should be at least some basic `pf` bosting when doing a search across all result types.

Reply:
We could map the phrase/query fields value for the "all result types" search to the empty key:

    'queryFields': {
        'DOCUMENT': 'title^10.0 name^10.0 ...',
        ...,
        '': '<the value to be used when there is no result type selected>'
    }

I think this will work because when there is no result type selected the ``$type`` variable (in ``SolrSearchMacros``) is null so writing this should work:

    #set ($boost = $solrConfig.queryFields.get("$!type"))

Using silent notation to get the empty string when ``$type`` is null. WDYT?

Reply:
Sounds like a good solution